Unexpose non-standard `resize: auto` value

nt1m opened the pull request:
#### b829c58e74144cfe9feb5c76e58f2a37d08d11b1
<pre>
Unexpose non-standard `resize: auto` value
<a href="https://bugs.webkit.org/show_bug.cgi?id=266614">https://bugs.webkit.org/show_bug.cgi?id=266614</a>
<a href="https://rdar.apple.com/120138995">rdar://120138995</a>

Reviewed by Anne van Kesteren.

Rename `resize: auto` to `resize: -internal-textarea-auto` and make it specific to UA stylesheets.

Also add some tests for the &quot;TextAreasAreResizable&quot; preference since it was previously untested.

* LayoutTests/fast/css/textarea-resizable-preference-disabled-expected.html: Added.
* LayoutTests/fast/css/textarea-resizable-preference-disabled.html: Added.
* LayoutTests/fast/css/textarea-resizable-preference-enabled-expected.html: Added.
* LayoutTests/fast/css/textarea-resizable-preference-enabled.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-ui/parsing/resize-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ui/parsing/resize-invalid.html:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/html.css:
(#endif):
* Source/WebCore/css/parser/CSSParserIdioms.cpp:
(WebCore::isValueAllowedInMode):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertResize):

Canonical link: <a href="https://commits.webkit.org/273035@main">https://commits.webkit.org/273035@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3910851dd7692580a69bd8414bf75ab1225a8823

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34022 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12814 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35990 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36657 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30835 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35092 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15198 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9955 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29886 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34522 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10837 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30358 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9455 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9560 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30378 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37965 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30904 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30701 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35668 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9689 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7619 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33570 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11474 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/29991 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7836 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10254 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10499 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (commented):
I wonder if given https://github.com/search?q=%22resize%3A+auto%22+language%3ACSS&type=code it would be better to standardize this value.

annevk reviewed (approved):
I see, it only impacting non-textarea element makes this less concerning. If you're prepared to deal with the fallout it seems reasonable to try.

nt1m committed:
Unexpose non-standard `resize: auto` value
https://bugs.webkit.org/show_bug.cgi?id=266614
rdar://120138995

Reviewed by Anne van Kesteren.

Rename `resize: auto` to `resize: -internal-textarea-auto` and make it specific to UA stylesheets.

Also add some tests for the "TextAreasAreResizable" preference since it was previously untested.

* LayoutTests/fast/css/textarea-resizable-preference-disabled-expected.html: Added.
* LayoutTests/fast/css/textarea-resizable-preference-disabled.html: Added.
* LayoutTests/fast/css/textarea-resizable-preference-enabled-expected.html: Added.
* LayoutTests/fast/css/textarea-resizable-preference-enabled.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-ui/parsing/resize-invalid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-ui/parsing/resize-invalid.html:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/html.css:
(#endif):
* Source/WebCore/css/parser/CSSParserIdioms.cpp:
(WebCore::isValueAllowedInMode):
* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertResize):

Canonical link: https://commits.webkit.org/273035@main
